File: src/main/entity/profiles/Person.java
package entity.profiles;

import entity.datafiles.Email;
import entity.datafiles.Name;
import entity.datafiles.Phone;

import java.io.Serializable;

public class Person implements ProfileType, Serializable {

    /** Stores all values as subclasses
     * Getters return all values as Strings
     */

    private final Name name;
    private final Phone phone;
    private final Email email;
    private final String username;


    public Person(Name name, Phone phone, Email email, String username) {
        this.name = name;
        this.phone = phone;
        this.email = email;
        this.username = username;
    }

    @Override
    public String getName() {
        return name.getFullName();
    }

    public String getPronouns() {
        return name.getPronouns();
    }

    @Override
    public String getPhone() {
        return phone.getPhone();
    }

    @Override
    public String getEmail() {
        return email.getEmail();
    }

    @Override
    public String getUsername() {
        return this.username;
    }

}

File: kard/lib/person.dart
class Person {

  String name;
  String phone;
  String email;
  String pronouns;
  String username;

  Person(
      this.name,
      this.phone,
      this.email,
      this.pronouns,
      this.username);

  Person.fromJson(Map<String, dynamic> json):
      name = json['name'] ?? '',
      phone = json['phone'] ?? '',
      email = json['email'] ?? '',
      pronouns = json['pronouns'] ?? '',
      username = json['username'] ?? '';
}